### ISSUE TITLE
Selecting clipped text from menu doesn't copy text

### ISSUE THREAD

**Issue by winchendonsprings** [cdown/clipmenu#18]
After I call clipmenu and select my new clipping it is not there when I past. It reverts to the last text copied.

dmenu 4.5
ubuntu 14.04

**Comment by cdown** [comment]
The only thing that clipmenu does when it comes to actually managing the clipboard is calling `xsel` or `xclip`, so if there is a problem, it's almost certainly related to that tool or your environment rather than clipmenu.

Here are some questions:

1. How are you launching clipmenu? (From a binding, from the terminal, etc.)
2. What do you mean by "reverts"? How do you know it reverts instead of never copying at all?
3. What is the output of `type xsel xclip`?
4. What version of clipmenu are you running? If not a version, which git revision?
5. Which clipboard are you pasting from? (`PRIMARY`, `SECONDARY`, `SELECTION`)
6. Does using xsel or xclip by themselves successfully update the clipboard without "reverting" it?

**Comment by winchendonsprings** [comment]
1. I start clipmenud with .xinit and I call clipmenu with a keybinding via xbindkeys.
2. You may be right, It may not be copying at all. Although the selected text is added to the clipmenu menu in dmenu.   
3. `xsel is /usr/bin/xsel`
`xclip is /usr/bin/xclip`
4. I've switched between master and sort_selections git branches
5. I dunno which clipboard. How can I tell?
6. xclip seems to work fine with `xclip -o` and xsel seems to work with `xsel -o` although it adds "%" at the end of the selected text.

Thanks for the quick reply!

**Comment by cdown** [comment]
1.) As long as the entries are showing up in dmenu, clipmenud is working. As for xbindkeys, it's possible that for some reason the environment is not right there. Could you show your xbindkeys configuration file? What is in `/tmp/q` if you bind something to `env > /tmp/q`?
2.) `clipmenud` populates the directory that dmenu gets its information from, so I think that's unrelated.
3.) clipmenu will prefer xsel, then. If `xsel` also appears to accept input on the command line, though, I doubt that's the issue.
4.) A recent master? sort_selections is out of date.
5.) Are you pasting by middle paste, or Ctrl-V?
6.) I mostly wanted to know about input, ie, does `xsel <<< foo` put "foo" on the clipboard, for example.

**Comment by winchendonsprings** [comment]
1. I have a very small xbindkeys file. It mostly sets buttons for volume and call dmenu. And this 
```
# view clipboard in dmenu                                                                                
"/home/ry/.clipmenu/clipmenu -nb '#4444dc' -nf '#bbbbbb' -sb '#444499' -sf '#ffffff'"                    
Mod4 + c
```
My /tmp/q shows my env if I put env setup in it 
4.  Yeah I've just done a fresh pull from master.
5. I can paste with either Ctrl-v or Shift-Insert
6. Yeah, xsel <<< foo puts foo in the clipboard.

After I bring up dmenu with clipmenu and I select the text I want to copy with the arrow buttons and the enter button. How should I paste the contents? It seems like that is my disconnect.

**Comment by cdown** [comment]
Re #1, the point was so that I could see your environment :-)

**Comment by winchendonsprings** [comment]
my [env](https://www.dropbox.com/s/6hctqyoyo3h5y50/ry.png?dl=0)

**Comment by cdown** [comment]
Does the code at https://github.com/cdown/clipmenu/tree/own_clipboard fix this for you?

**Comment by winchendonsprings** [comment]
No luck with that branch. 

To clarify in order again. 
- I select text with mouse. It is copied to the primary clipboard.
- I paste with Shift-Insert and the text I had just selected was pasted.
- I open clipmenu/dmenu and use arrow keys to go down to a bit of selected text that is older than the most current selection and choose it with the Enter key.
- I paste with Shift-Insert and still the most current selected text is pasted, rather than what I had just selected.

**Comment by cdown** [comment]
Does copying and pasting using the `CLIPBOARD` clipboard work as expected, ie. you select something and you can paste it into (for example) a web browser with Ctrl-V?

**Comment by winchendonsprings** [comment]
Highlighting text, plus using Ctrl-C works as expected with Ctrl-V or choosing paste from a application menu. 

I dunno if this helps, but Parcellite works as expected. Highlight any text, choose any item to paste from Parcellite pop-up menu paste.

**Comment by cdown** [comment]
I just realised, you said you tried the `master` branch. Could you please try the `develop` branch? I'm wondering if this is the same mysterious issue as in #11.

**Comment by winchendonsprings** [comment]
Yes, develop branch fixed the problem. 

I am also using i3 similar to https://github.com/cdown/clipmenu/pull/11 but I call my keys from xbindkeys rather than the i3 config

**Comment by cdown** [comment]
Awesome! I've released version 1.1.0 with that fix in. Thanks for the report :-)
